course/src/main/java/in/testpress/course/util/ExoPlayerUtil.java

```diff
@@ -704,15 +711,29 @@ public void onSuccess(VideoAttempt videoAttempt) {
                         if (videoAttempt != null) {
                             updateVideoWatchedPercentage(videoAttempt);
                         }
-                        errorOnVideoAttemptUpdate = false;
+                        lastApiCallTime = System.currentTimeMillis() / 1000;
+                        throttleTimeRemaining = 0;
                     }
 
                     @Override
                     public void onException(TestpressException exception) {
                         if (videoWatchDataRepository != null) {
                             videoWatchDataRepository.save(content, getVideoAttemptParameters());
                         }
-                        errorOnVideoAttemptUpdate = true;
+                        if (exception.isTooManyRequest()){
+                            TestpressErrorDetail testpressErrorDetail = exception.getErrorBodyAs(exception.getResponse(),TestpressErrorDetail.class);
+                            throttleTimeRemaining = extractThrottleTime(testpressErrorDetail.getDetail());
```

Comment:
Can this be extracted somewhere, something like exception.getThrottleTime so that we can easily use this anywhere